api/src/mutations/drop.rs

```diff
@@ -241,6 +243,138 @@ impl Mutation {
             drop: Drop::new(drop_model, collection_model),
         })
     }
+
+    /// This mutation allows updating a drop and it's associated collection by ID.
+    /// It returns an error if it fails to reach the database, emit update events or assemble the on-chain transaction.
+    /// Returns the `CreateDropPayload` object on success.
+    pub async fn edit_drop(
+        &self,
+        ctx: &Context<'_>,
+        input: EditDropInput,
+    ) -> Result<CreateDropPayload> {
+        let EditDropInput {
+            drop,
+            price,
+            start_time,
+            end_time,
+            seller_fee_basis_points,
+            metadata_json,
+        } = input;
+
+        if price.is_none()
+            && start_time.is_none()
+            && end_time.is_none()
+            && seller_fee_basis_points.is_none()
+            && metadata_json.is_none()
+        {
+            return Err(Error::new("please select atleast one field to update"));
+        }
```

Comment:
Ok so this keeping all the previous settings? We should call this patch_drop then to signal to the caller they just need to alter what they want changed.

So input goes to PatchDropInput and PatchDropResponse.

Reply:
```suggestion
        if vec![price, start_time, end_time, seller_fee_basis_points, metadata_json].iter().all(|i| i.is_none())
        {
            return Err(Error::new("please select atleast one field to update"));
        }
```

nb but an option

Reply:
but all elements in vector should have same data type

Reply:
> Ok so this keeping all the previous settings? We should call this patch_drop then to signal to the caller they just need to alter what they want changed.
> 
> So input goes to PatchDropInput and PatchDropResponse.

Do we want to send all the fields that can be updated as input? I thought we need to send the field which needs to be updated.

Reply:
Either. Edit would be a put operation which replaces all fields on the record making the caller send over all data that can change. Just sending what you want updated is a PATCH operation which is normally friendly so happy to go with this but should name the mutation patch_drop to make clear that the caller only needs to send the fields they want to update.